inc/ThirdParty/Plugins/Optimization/AMP.php

```diff
@@ -24,18 +31,18 @@ class AMP implements Subscriber_Interface {
 	/**
 	 * Constructor
 	 *
-	 * @param Options_Data $options WP Rocket Options instance.
+	 * @param Options_Data $options        WP Rocket Options instance.
+	 * @param Subscriber   $cdn_subscriber WP Rocket CDN Subscriber.
 	 */
-	public function __construct( Options_Data $options ) {
-		$this->options = $options;
+	public function __construct( Options_Data $options, $cdn_subscriber ) {
```

Comment:
The subscriber implements an interface. We can type hint to the interface instead of the concrete like this:


```php

use WP_Rocket\Event_Management\Subscriber_Interface;

// code left out for brevity.

/**
 * Instance of CDN Subscriber.
 *
 * @var Subscriber_Interface
 */
private $cdn_subscriber;

/**
 * Constructor
 *
 * @param Options_Data $options WP Rocket Options instance.
 * @param Options_Data $options        WP Rocket Options instance.
 * @param Subscriber_Interface   $cdn_subscriber Instance of CDN Subscriber.
 */
public function __construct( Options_Data $options, Subscriber_Interface $cdn_subscriber ) {
```

Why? It allows us to swap out the concrete for testing or some future use case.